### ISSUE TITLE
Archiving fails with "Multiple commands produce '.../MyPod.framework'"

### ISSUE THREAD

**Issue by imre-chroncsik** [CocoaPods/CocoaPods#11016]
* [x] I've read and understood the [*CONTRIBUTING* guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md). 🌈

# Report

With certain setups, archiving a CocoaPods-generated workspace fails, reporting "Multiple commands produce '.../MyPod.framework'". 

Might be related to https://github.com/CocoaPods/CocoaPods/issues/10980, although the error reported is slightly different, but the project / pod setup is similar. 

*	MyPod has two subspecs, Core and Optional, Core being the default.  
*	The app has two targets, they both add MyPod, only one of them adds MyPod/Optional. 
*	`pod install` now creates a `MyPod` and a `MyPod.default-Optional` target. Optional.swift is a member only of the latter, Core.swift is a member of both. 
*	The project builds, but archiving it fails with "Multiple commands produce '.../MyPod.framework'"
*	If in the app's Podfile both app targets add the Optional subspec too, then `pod install` creates only a single `MyPod` target (which also includes Optional.swift), and archiving succeeds. 

## What did you do?

`pod install`, then Xcode Product > Archive. 

## What did you expect to happen?

Xcode to correctly produce an archive using the workspace generated by CocoaPods. Or, of it's a setup issue, then `pod install` to report an error. 

## What happened instead?

Xcode reported an error "Multiple commands produce '.../MyPod.framework'". 

### Stack

(Note I've removed the URLs to some private repos, because I'm not sure about corporate rules for listing them publicly. They have nothing to do with the attached sample anyway.)

```
   CocoaPods : 1.11.2
        Ruby : ruby 2.6.3p62 (2019-04-16 revision 67580) [universal.x86_64-darwin20]
    RubyGems : 3.0.3
        Host : macOS 11.6 (20G165)
       Xcode : 13.0 (13A233)
         Git : git version 2.30.1 (Apple Git-130)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib
Repositories : cocoapods - git - https://github.com/CocoaPods/Specs.git @ 1bd6cdcd8358ed2971937846001c35f978b0429e

               <removed>

               trunk - CDN - https://cdn.cocoapods.org/
               <removed>

               <removed>

               <removed>

               <removed>

```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.4
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.1.0
cocoapods-trunk       : 1.5.0
cocoapods-try         : 1.2.0
```

### Podfile

```ruby
platform :ios, '12.0'

target 'podtest' do
  use_frameworks!
  pod 'MyPod', :path => '../mypod'
  pod 'MyPod/Optional', :path => '../mypod'
end

target 'podtest-2' do
  use_frameworks!
  pod 'MyPod', :path => '../mypod'

  # if this line is commented out, then archiving fails.
  # if added back, the build succeeds.
#  pod 'MyPod/Optional', :path => '../mypod'

end
```

## Project that demonstrates the issue

https://u.pcloud.link/publink/show?code=XZw3UEXZ0uycR0pTYmJlUf5Pxm9wjhd38urk

**Comment by JodusNodus** [comment]
@imre-chroncsik We're having the same issue, have you found a solution?

**Comment by imre-chroncsik** [comment]
@JodusNodus Not really. My workaround is to add the Optional subspec to all targets, even to those where it isn't really used. Works in my specific use case, but probably not always.

**Comment by sansui-orz** [comment]
> @JodusNodus Not really. My workaround is to add the Optional subspec to all targets, even to those where it isn't really used. Works in my specific use case, but probably not always.

Yes, It really work for me.

中文具体修改可以看这里：https://segmentfault.com/q/1010000043614798
